indico/modules/events/contributions/models/contributions.py

```diff
@@ -47,8 +48,31 @@ def _get_next_friendly_id(context):
     return increment_and_get(Event._last_friendly_contribution_id, Event.id == event_id)
 
 
+class CustomFieldsMixin:
+    """Store information about contribution custom fields."""
+
+    def save_custom_fields(self, custom_field_values):
+        for custom_field in self.event_new.contribution_fields:
+            data = custom_field_values.get('custom_{}'.format(custom_field.id))
+            if data:
+                ContributionFieldValue(contribution_field=custom_field, contribution=self, data=data)
+
+    def set_custom_field(self, field_id, field_value):
+        field = self.get_field(field_id)
+        if field:
```

Comment:
```
fv = self.get_field(field_id)  # get_field_value?! see my other comment
if not field:
    fv = ContributionFieldValue(contribution_field=self.event_new.get_contribution_field(field_id))
    self.field_values.append(fv)
fv.data = field_value
```

- only one place where the value it set
- it makes it clearer that you don't just create an object without it being connected to anything (your approach works too but instantiating a class without assigning it to something looks wrong to me)

also, this will eventually be used for abstracts